### DISCUSSION TITLE
Nested filters

### DISCUSSION THREAD

**Discussion by jesseparot** [BaseQL/baseql-docs#4] [Ideas]
Hi,

Great product idea. Was wondering why nested filters such as this example from the Hasura website don't work? 
It's quite disappointing as it would be one of the strengths graph would bring to airtable 

`query {
     articles (where: {author: {name: {_eq: "Sidney"}}}) {
       id
       title
     }
   }`

Thanks

**Comment by benoror** [comment]
Hey @jesseparot ! Thanks for reaching out, that's a great idea! Just to clarify, in that example is `author` a relationship to another table? Or another kind of calculated value?

**Reply by jesseparot** [reply]
@benoror Maybe explaining what I'm trying to achieve will make it easier 😅  Basically I'm building a no/low-code app from which I query content based on the user's preferences so my tables in Airtable are organised as such:
Users: UID - email - sports preferences (linked record) - formats preferences (linked record) - tags preferences (linked record)
Content: Title - Link - sports (linked records) - formats (linked records) - tags (linked records)
Sports: Name - Content - Users
Formats: Name - Content - Users
Tags: Name - Content - Users

What I wanted to achieve was this:

```graphQL
query($userID: String){
  content(_filter: {
    _or: [
      _and: [
        {content: {sports: {users: {_contains: $userID}}}},
        {content: {formats: {users: {_contains: $userID}}}}
      ],
      {content: {sports: {users: {_contains: $userID}}}}
    ]}) 
{
  title
  link
}

```

It's probably not the exact query as I'm new to GraphQL but it would be an easier way for me, the alternative being making multiple call through Integromat.

Hope it's not too messy. Appreciate you responding so quickly.

**Reply by benoror** [reply]
How about filtering starting from the other way around, and querying for the nested values? In the first example, instead finding `articles` first, starting with `author`'s instead:

```graphql
query {
  author (_filter: { name: {_eq: "Sidney"} }) {
    articles {
      id
      title
    }
  }
}
```

On the second example, if I am understanding well, would be something like:

```graphql
query($userID: String) {
  users (_filter: { id: {_eq: $userID} }) {
    sports {
      content {
          title
          link
      }
    }
    formats {
      content {
          title
          link
      }
    }
  }
}
```

Hope it makes sense. Taking a look at the tables would help as well. Feel free to email at support@baseql.com
